src/clib/lib/job_queue/slurm_driver.cpp

```diff
@@ -175,26 +161,25 @@ static std::string load_stdout(const char *cmd,
 
 static std::vector<std::string> split_string(const std::string &string_value) {
     std::vector<std::string> strings;
-    std::size_t offset = string_value.find_first_not_of(", ");
-    while (offset != std::string::npos) {
-        auto item_end = string_value.find_first_of(", ", offset);
-        strings.push_back(string_value.substr(offset, item_end - offset));
-        offset = string_value.find_first_not_of(", ", item_end);
-    }
+    std::stringstream ss(string_value);
+    std::string intermediate;
+
+    while (getline(ss, intermediate, ','))
+        strings.push_back(
+            intermediate.substr(intermediate.find_first_not_of(' ')));
+
     return strings;
 }
 
 template <typename C> static std::string join_string(const C &strings) {
-    const std::string sep = ",";
-    std::string full_string;
-    bool first = true;
-    for (const auto &s : strings) {
-        if (!first)
-            full_string += sep;
-        full_string += s;
-        first = false;
+    std::string str;
+
+    for (const auto &key : strings) {
+        str.append(",");
+        str.append(key);
     }
-    return full_string;
+
+    return str.substr(str.find_first_not_of(','));
```

Comment:
Maybe keep the `first` boolean value instead of using substr?

Reply:
Fine. I find this more elegant, but they are almost alike.